app/javascript/src/components/Player.tsx

```diff
@@ -203,7 +205,7 @@ class Player extends React.Component<PlayerProps, PlayerState> {
             {this.renderChoices(node)}
           </div>
         </div>
-        {this.renderDebugger()}
+        {this.renderDebugSection()}
```

Comment:
👍 

Unrelated question: why does this component have render side effects? (i.e. `this.hideSourceButton()`. Is this because the element with a class of `-source-link` is outside of this component's render tree?

Render _should_ be idempotent. An alternative would be to use `componentDidUpdate`.

Reply:
Ok, I did not write the code around the source button, but I just looked into it a bit and it seems like the Source button with the `-source-link` class is coming from the `show.html.erb` file. I could fairly easily move that button's creation into the `Player.tsx` file if you think that would be better practice though.